contributors/devel/sig-testing/flaky-tests.md

```diff
@@ -140,34 +158,242 @@ release-blocking flakes. Therefore we have the following guidelines:
+[stress command](https://godoc.org/golang.org/x/tools/cmd/stress). Install it
+with this command: 
+
+```sh
+go get golang.org/x/tools/cmd/stress
 ```
 
-Then build your test binary
+Then build your test binary:
 
+```sh
+go test ./pkg/kubelet/config -race -c
 ```
-$ go test -c -race
+
+Then run it under stress:
+
+```sh
+stress ./config.test -test.run TestInvalidPodFiltered
 ```
 
-Then run it under stress
+The stress command runs the test binary repeatedly, reporting when it fails. It
+will periodically report how many times it has run and how many failures have
+occurred. 
+
+You should see output like this:
 
 ```
-$ stress ./package.test -test.run=FlakyTest
+411 runs so far, 0 failures
+/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200825T115041-341977266
+--- FAIL: TestInvalidPodFiltered (0.00s)
+    config_test.go:126: Expected no update in channel, Got types.PodUpdate{Pods:[]*v1.Pod{(*v1.Pod)(0xc00059e400)}, Op:1, Source:"test"}
+FAIL
+ERROR: exit status 1
+815 runs so far, 1 failures
 ```
 
-It runs the command and writes output to `/tmp/gostress-*` files when it fails.
-It periodically reports with run counts. Be careful with tests that use the
-`net/http/httptest` package; they could exhaust the available ports on your
-system!
+Be careful with tests that use the `net/http/httptest` package; they could
+exhaust the available ports on your system!
+
+### Reproducing flakes using integration tests
+
+Integration tests run similarly to unit tests, but they almost always expect a
+running `etcd` instance. You should already have `etcd` installed if you have
+followed the instructions in the [Development Guide](../development.md). Run
+`etcd` in another shell window or tab.
+
+Compile your integration test using a command like this:
+
+```sh
+go test -c -race ./test/integration/endpointslice
+```
+
+And then stress test the flaky test using the `stress` command:
+
+```sh
+stress ./endpointslice.test -test.run TestEndpointSliceMirroring
+```
 
-# Hunting flaky unit tests in Kubernetes
+For an example of a failing or flaky integration test,
+[read this issue](https://github.com/kubernetes/kubernetes/issues/93496#issuecomment-678375312). 
+
+Sometimes, but not often, a test will fail due to timeouts caused by
+deadlocks. This can be tracked down by stress testing an entire package. The way
+to track this down is to stress test individual tests in a package. This process
+can take extra effort. Try following these steps:
+
+1. Run each test in the package individually to figure out the average runtime.
+2. Stress each test individually, bounding the timeout to 100 times the average run time.
+3. Isolate the particular test that is deadlocking.
+4. Add debug output to figure out what is causing the deadlock.
+
+Hopefully this can help narrow down exactly where the deadlock is occurring,
+revealing a simple fix!
+
+## Deflaking e2e tests
+
+A flaky [end-to-end (e2e) test](e2e-tests-kubetest2.md) offers its own set of
```

Comment:
I think it would be better to link the original doc since it includes info about test tags, and kubetest2 isn't yet heavily used by the majority of release-blocking or merge-blocking jobs for kubernetes/kubernetes
```suggestion
A flaky [end-to-end (e2e) test](e2e-tests.md) offers its own set of
```